### PR TITLE
Fixed #3416 - templates with class phpdocs

### LINKED ISSUE #3416: Broken parsing of '@template of I' followed by a simple comment
When `@template of` is followed by a simple comment on a next line, psalm fails to stop parsing the template tag, and continues to the next line, making the comment part of `of I`

https://psalm.dev/r/f5aa676997

Without 'of I' everything is parsed properly

https://psalm.dev/r/74b00ac705

Same applies for at least one more tag -  `@extends`

### PR DESCRIPTION
**Pull request by elnoro**
This is a partial fix for https://github.com/vimeo/psalm/issues/3416

### CODE REVIEW

File: src/Psalm/Internal/PhpVisitor/Reflector/ClassLikeDocblockParser.php [hunk 1]
```diff
@@ -53,7 +53,8 @@ public static function parse(
         $info = new ClassLikeDocblockComment();
 
         if (isset($parsed_docblock->combined_tags['template'])) {
-            foreach ($parsed_docblock->combined_tags['template'] as $offset => $template_line) {
+            foreach ($parsed_docblock->combined_tags['template'] as $offset => $template_tag) {
+                $template_line = preg_replace('/\n.*/m', '', $template_tag);
```
Comment (weirdan) [inline src/Psalm/Internal/PhpVisitor/Reflector/ClassLikeDocblockParser.php:57]: Wouldn't it break more complex structures like:
```php
/**
 * @template T of array{
 *    a: string,
 *    b: int
 * }
 */
```

**Comment by elnoro** [comment]
You are right, it will probably break it. I’ll see if I can find a better way to do this

**Comment by elnoro** [comment]
Closed this since I am not sure what is the best way to resolve this without breaking BC. 

@weirdan do you think it's worth to add a test with a complex structure like the one in your example to ClassTemplateTest.php? This should save you some time next time someone attempts a similar solution.

**Comment by weirdan** [comment]
> do you think it's worth it to add a test with a complex structure 

Feel free to PR that.

**Pull request closed without merging**